### PR TITLE
feat(content-rich html parsing): add paragraphs LTR/RTL direction support

### LINKED ISSUE #2544: add support for multiple translations entries mixing LTR and RTL text
Right now there is no way to include multiple lang texts when composing, the lang in the composer will apply to the full text.

The new proposal is about to support LTR/RTL text using multiple paragraphs using `bdi` or just adding `dir="auto"` (it also works).

For example, if anyone wants to include english and some arabic translation (or viceversa), we can enable a new flag in the composer options to include inside any `p` a new `bdi` child (it is just an span) with the text (rn the text will be just added to the `p`).

<details>
<summary>Here an example using English + Arabic translation using &lt;bdi></summary>
<br/>
<p><bdi>I like how the arabic dual form is supported. The last one is mistranslated and I'm trying to find how to fix it. Also, the form for 0 needs to be fixed.</bdi></p>
<p><bdi>أنا أحب الطريقة التي يتم بها دعم النموذج المزدوج العربي. تمت ترجمة الكلمة الأخيرة بشكل خاطئ وأحاول العثور على كيفية إصلاحها. أيضًا، يجب إصلاح نموذج 0.</bdi></p>
</details>

<details>
<summary>Previous example without &lt;bdi> (gh adding dir="auto")</summary>
<br/>
<p>I like how the arabic dual form is supported. The last one is mistranslated and I'm trying to find how to fix it. Also, the form for 0 needs to be fixed.</p>
<p>أنا أحب الطريقة التي يتم بها دعم النموذج المزدوج العربي. تمت ترجمة الكلمة الأخيرة بشكل خاطئ وأحاول العثور على كيفية إصلاحها. أيضًا، يجب إصلاح نموذج 0.</p>
</details>

Here Elk toot (notice the final dot is wrong, should be after the 0 in RTL):

imagen

The `content-rich` wrapper will use `lang="en"` in previous example:

imagen

/cc @patak-dev @danielroe check the source code in previous details and the source page

### PR DESCRIPTION
**Pull request by userquin**
Included a new `addBdiParagraphs` function and calling it in the `transformParagraphs` callback.

It seems we only need to cover 2 cases:
- mixed content paragraphs, text, icons: adding `dir="auto"` to the root `p` works
- paragraphs with text + `br` nodes only: replace root `p`  with `div`, wrap text nodes with `p dir="auto"` and remove any `br`

I need to review a few things, awaiting some inputs from `bebatjof`.

First case (https://deploy-preview-2545--elk-zone.netlify.app/universeodon.com/@userquin@webtoo.ls/111706258253094733):

imagen

Second case (https://deploy-preview-2545--elk-zone.netlify.app/universeodon.com/@userquin@webtoo.ls/111706237164074626):

imagen

closes #2544

### CODE REVIEW

File: composables/content-parse.ts [hunk 1]
```diff
@@ -524,10 +524,60 @@ function transformMarkdown(node: Node) {
   return _markdownProcess(node.value)
 }
 
+function addBdiParagraphs(node: Node) {
+  if (node.name === 'p' && node.children && node.children.length > 1) {
+    const split = node.children.every((child: Node) => child.type === TEXT_NODE || child.name === 'br')
```
Comment (userquin) [inline composables/content-parse.ts:529]: Exclude also paraghaps with dir attr

File: composables/content-parse.ts [hunk 2]
```diff
@@ -524,10 +524,60 @@ function transformMarkdown(node: Node) {
...
+      node.attributes.dir = 'auto'
+      return node
+    }
+
+    // convert <p>some test<br/>some text...<br/>some text...</p>
+    // to <div><p dir="auto">some text</p><p dir="auto">some text...</p><p dir="auto">some text...</p></div>
+    node.name = 'div'
+    const children = node.children.splice(0) as Node[]
+    let child: Node
+    for (let i = 0; i < children.length; i++) {
```
Comment (userquin) [inline composables/content-parse.ts:541]: replace this with of

File: composables/content-parse.ts [hunk 3]
```diff
@@ -524,10 +524,60 @@ function transformMarkdown(node: Node) {
...
+      child = children[i]
+      if (child.name === 'p') {
```
Comment (userquin) [inline composables/content-parse.ts:543]: this will never be true, children are only text or br

File: composables/content-parse.ts [hunk 4]
```diff
@@ -524,10 +524,54 @@ function transformMarkdown(node: Node) {
...
+    const split = node.children.every((child: Node) => child.type === TEXT_NODE || child.name === 'br')
+    if (!split) {
+      // adding dir="auto" to the root paragraph just works
+      node.attributes.dir = 'auto'
+      return node
+    }
+
+    // convert <p>some test<br/>some text...<br/>some text...</p>
+    // to <div><p dir="auto">some text</p><p dir="auto">some text...</p><p dir="auto">some text...</p></div>
+    node.name = 'div'
```
Comment (userquin) [inline composables/content-parse.ts:538]: Revert this

**Comment by patak-dev** [comment]
This is breaking spacing between paragraphs:

working: https://deploy-preview-2642--elk-zone.netlify.app/universeodon.com/@elkdev/112039559654828608
broken: https://deploy-preview-2545--elk-zone.netlify.app/universeodon.com/@elkdev/112039559654828608

Maybe we could add `auto` only to `p` and avoid modifying nodes. It would still be a big improvement and it is safer to merge.

**Comment by userquin** [comment]
auto in p not working iirc, we can update the style to remove the y-padding...

File: composables/content-parse.ts [hunk 5]
```diff
@@ -526,39 +526,8 @@ function transformMarkdown(node: Node) {
...
-        loc: node.loc,
-        type: ELEMENT_NODE,
-        attributes: { dir: 'auto' },
-        children: [child],
-      }
-      child.parent = p
-      node.children.push(p)
-    }
+    node.attributes.dir = 'auto'
+    return node
```
Comment (patak-dev) [inline composables/content-parse.ts:530]: We can also remove this line
```suggestion
```

Comment (userquin) [reply composables/content-parse.ts]: done

File: composables/content-parse.ts (hunk 4 above)
Comment (userquin) [reply composables/content-parse.ts]: reverted to a simplified version: adding only dir auto to the paragraph

**Review by patak-dev — APPROVED** [review]

**Pull request merged**